chainextensions/dapps-staking/impl/src/lib.rs

```diff
@@ -0,0 +1,489 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+use sp_runtime::{
+    traits::{Saturating, Zero},
+    DispatchError,
+};
+
+use chain_extension_traits::ChainExtensionExec;
+use chain_extension_types::{
+    Contract, DSError, DappsStakingAccountInput, DappsStakingEraInput, DappsStakingNominationInput,
+    DappsStakingValueInput,
+};
+use codec::{Decode, Encode};
+use frame_support::traits::{Currency, Get};
+use frame_system::RawOrigin;
+use pallet_contracts::chain_extension::{Environment, Ext, InitState, SysConfig, UncheckedFrom};
+use pallet_dapps_staking::{RewardDestination, WeightInfo};
+use sp_std::marker::PhantomData;
+
+type BalanceOf<T> = <<T as pallet_dapps_staking::Config>::Currency as Currency<
+    <T as frame_system::Config>::AccountId,
+>>::Balance;
+
+enum DappsStakingFunc {
+    CurrentEra,
+    UnbondingPeriod,
+    EraRewards,
+    EraStaked,
+    StakedAmount,
+    StakedAmountOnContract,
+    ReadContractStake,
+    Register,
+    BondAndStake,
+    UnbondAndUnstake,
+    WithdrawUnbonded,
+    ClaimStaker,
+    ClaimDapp,
+    SetRewardDestination,
+    NominationTransfer,
+}
+
+impl TryFrom<u32> for DappsStakingFunc {
+    type Error = DispatchError;
+
+    fn try_from(value: u32) -> Result<Self, Self::Error> {
+        match value {
+            1 => Ok(DappsStakingFunc::CurrentEra),
+            2 => Ok(DappsStakingFunc::UnbondingPeriod),
+            3 => Ok(DappsStakingFunc::EraRewards),
+            4 => Ok(DappsStakingFunc::EraStaked),
+            5 => Ok(DappsStakingFunc::StakedAmount),
+            6 => Ok(DappsStakingFunc::StakedAmountOnContract),
+            7 => Ok(DappsStakingFunc::ReadContractStake),
+            8 => Ok(DappsStakingFunc::Register),
+            9 => Ok(DappsStakingFunc::BondAndStake),
+            10 => Ok(DappsStakingFunc::UnbondAndUnstake),
+            11 => Ok(DappsStakingFunc::WithdrawUnbonded),
+            12 => Ok(DappsStakingFunc::ClaimStaker),
+            13 => Ok(DappsStakingFunc::ClaimDapp),
+            14 => Ok(DappsStakingFunc::SetRewardDestination),
+            15 => Ok(DappsStakingFunc::NominationTransfer),
+            _ => Err(DispatchError::Other(
+                "DappsStakingExtension: Unimplemented func_id",
+            )),
+        }
+    }
```

Comment:
Why not adopt the same approach as EVM does regarding the function identifier?

Reply:
@Dinonard can you share the code line of it?